### ISSUE TITLE
"break when true" causes IndexErrror

### ISSUE THREAD

**Issue by dawkot** [crystal-lang/crystal#6508]
```crystal
loop do
    break when true
end
```
Full error message when compiled on Windows using WSL (with Ubuntu):

> Index out of bounds (IndexError)
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
>   from ???
> Error: you've found a bug in the Crystal compiler. Please open an issue, including source code that will allow us to reproduce the bug: https://github.com/crystal-lang/crystal/issues

**Comment by Sija** [comment]
Please include a proper backtrace. Millions of `from ???` lines tell us nothing.

**Comment by z64** [comment]
@Sija that is what it prints https://carc.in/#/r/4p9i

**Comment by asterite** [comment]
The backtrace doesn't matter much, this is just a simple parser bug

**Comment by Sija** [comment]
@z64 Yeah, that's why you should do it on your local machine and not carc.in.

**Comment by z64** [comment]
@Sija Yeah my bad, I just misunderstood your original question. :+1:
